js_modules/dagster-ui/packages/ui-core/src/assets/asset-checks/ExecuteChecksButton.tsx

```diff
@@ -45,6 +46,14 @@ export const ExecuteChecksButton = ({
     ? 'No checks are defined on this asset.'
     : '';
 
+  const {
+    featureContext: {canSeeExecuteChecksAction},
+  } = useContext(CloudOSSContext);
+
+  if (!canSeeExecuteChecksAction) {
+    return null;
+  }
+
```

Comment:
This looks good to me -- agree that it feels a bit gross to have this layered on top of the permissions system (and a bit confusing because the permission disables the button, and this fully hides the button), but especially because of the disable vs. hide split, it does probably make sense to implement it this way.